azure/services/async/async.go

```diff
@@ -85,16 +85,14 @@ func CreateResource(ctx context.Context, scope FutureScope, client Creator, spec
 	// No long running operation is active, so create the resource.
 	scope.V(2).Info("creating resource", "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
 	result, sdkFuture, err := client.CreateOrUpdateAsync(ctx, spec)
-	if err != nil {
-		if sdkFuture != nil {
-			future, err := converters.SDKToFuture(sdkFuture, infrav1.PutFuture, serviceName, resourceName, rgName)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to create resource %s/%s (service: %s)", rgName, resourceName, serviceName)
-			}
-			scope.SetLongRunningOperationState(future)
-			return nil, azure.WithTransientError(azure.NewOperationNotDoneError(future), retryAfter(sdkFuture))
+	if sdkFuture != nil {
```

Comment:
@karuppiah7890 please take a look. I believe this addresses your comment in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1684#discussion_r719192451. I think it makes sense to not require an error (that way this will work whether an error is returned or not). However, I still think we shouldn't silence the error returned by `WaitForCompletionRef` in the clients since eventually it might return a different error besides context deadline exceeded.